File: C++/024. Swap Nodes in Pairs.cpp
#include <iostream>

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode *res = new ListNode(0);
        res->next = head;
        ListNode *p = res;
        while (p->next != nullptr && p->next->next != nullptr) {
            ListNode *tmp = p->next;
            p->next = p->next->next;
            tmp->next = p->next->next;
            p->next->next = tmp;
            p = p->next->next;
        }
        return res->next;
    }
};

File: Go/024. Swap Nodes in Pairs.go
package main

type ListNode struct {
    Val int
    Next *ListNode
}

func swapPairs(head *ListNode) *ListNode {
	res := &ListNode{Next:head}
	p := res
	for p.Next != nil && p.Next.Next != nil {
		tmp := p.Next
		p.Next = p.Next.Next
		tmp.Next = p.Next.Next
		p.Next.Next = tmp
		p = p.Next.Next
	}
	return res.Next
}